3p/integration.js

```diff
@@ -453,6 +454,21 @@ function triggerRenderStart(opt_data) {
 }
 
 /**
+ * See readme for window.context.getHtml
+ * @param {!string} selector - CSS selector of the node to take content from
+ * @param {!Array<string>} attributes - tag attributes to be left in the stringified HTML
+ * @param {!Function} callback
+ */
+function getHtml(selector, attributes, callback) {
+  nonSensitiveDataPostMessage('get-html', {selector, attributes});
+
+  const unlisten = listenParent(window, 'get-html-result', data => {
```

Comment:
There's a delicate race condition if this is called multiple times (first response will fulfill all waiting). We could pass back and forth a global counter and fulfill when the correct number is returned, or just throw when a duplicate request is made before the last one has fulfilled.